lua/entities/gmod_wire_expression2/core/custom/cfc_vector2_NumberOperands.lua

```diff
@@ -0,0 +1,537 @@
+    }
+end
+e2function vector2 vector2:acosvec2()
+    return {
+        acos(this[1]) * rad2deg,
+        acos(this[2]) * rad2deg
+    }
+end
+
+e2function number acos(vector2 vectah)
+    local hypo = sqrt(vectah[1]^2+vectah[2]^2)
+    return acos(vectah[1]/hypo) * rad2deg
+end
+e2function number vector2:acos()
+    local hypo = sqrt(this[1]^2+this[2]^2)
+    return acos(this[1]/hypo) * rad2deg
+end
+
+e2function vector2 asinvec2(vector2 vectah)
+    return {
+        asin(vectah[1]) * rad2deg,
+        asin(vectah[2]) * rad2deg
+    }
+end
+e2function vector2 vector2:asinvec2()
+    return {
+        asin(this[1]) * rad2deg,
+        asin(this[2]) * rad2deg
+    }
+end
+
+e2function number asin(vector2 vectah)
+    local hypo = sqrt(vectah[1]^2+vectah[2]^2)
+    return asin(vectah[2]/hypo) * rad2deg
+end
+e2function number vector2:asin()
+    local hypo = sqrt(this[1]^2+this[2]^2)
+    return asin(this[2]/hypo) * rad2deg
+end
+
+e2function vector2 atanvec2(vector2 vectah)
+    return {
+        atan(vectah[1]) * rad2deg,
+        atan(vectah[2]) * rad2deg
+    }
+end
+e2function vector2 vector2:atanvec2()
+    return {
+        atan(this[1]) * rad2deg,
+        atan(this[2]) * rad2deg
+    }
+end
+
+e2function number atan(vector2 vectah)
+    return atan(vectah[2]/vectah[1]) * rad2deg
+end
+e2function number vector2:atan()
+    return atan(this[2]/this[1]) * rad2deg
+end
+
+e2function vector2 cos(vector2 vectah)
+    return {
+        cos(vectah[1]*deg2rad),
+        cos(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:cos()
+    return {
+        cos(this[1]*deg2rad),
+        cos(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 sec(vector2 vectah)
+    return {
+        1/cos(vectah[1]*deg2rad),
+        1/cos(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:sec()
+    return {
+        1/cos(this[1]*deg2rad),
+        1/cos(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 sin(vector2 vectah)
+    return {
+        sin(vectah[1]*deg2rad),
+        sin(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:sin()
+    return {
+        sin(this[1]*deg2rad),
+        sin(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 csc(vector2 vectah)
+    return {
+        1/sin(vectah[1]*deg2rad),
+        1/sin(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:csc()
+    return {
+        1/sin(this[1]*deg2rad),
+        1/sin(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 tan(vector2 vectah)
+    return {
+        tan(vectah[1]*deg2rad),
+        tan(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:tan()
+    return {
+        tan(this[1]*deg2rad),
+        tan(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 cot(vector2 vectah)
+    return {
+        1/tan(vectah[1]*deg2rad),
+        1/tan(vectah[2]*deg2rad)
+    }
+end
+e2function vector2 vector2:cot()
+    return {
+        1/tan(this[1]*deg2rad),
+        1/tan(this[2]*deg2rad)
+    }
+end
+
+e2function vector2 rcos(vector2 vectah)
```

Comment:
```suggestion
e2function vector2 cosr(vector2 vectah)
```

Same goes with the following functions.

Reply:
Done